homeassistant/components/switch/light.py

```diff
@@ -45,17 +47,25 @@ async def async_setup_platform(
 ) -> None:
     """Initialize Light Switch platform."""
     async_add_entities(
-        [LightSwitch(cast(str, config.get(CONF_NAME)), config[CONF_ENTITY_ID])], True
+        [
+            LightSwitch(
+                cast(str, config.get(CONF_NAME)),
+                config[CONF_ENTITY_ID],
+                cast(str, config.get(CONF_UNIQUE_ID)),
+            )
+        ],
+        True,
     )
 
 
 class LightSwitch(LightEntity):
     """Represents a Switch as a Light."""
 
-    def __init__(self, name: str, switch_entity_id: str) -> None:
+    def __init__(self, name: str, switch_entity_id: str, unique_id: str) -> None:
         """Initialize Light Switch."""
         self._name = name
         self._switch_entity_id = switch_entity_id
+        self._unique_id = unique_id
```

Comment:
Can we use the `switch_entity_id` as unique id?

Reply:
I guess we can. But we will lose customization if the end user decide to change the entity wrapped by this switch, for instance migrating to the new z-wave integration.